dali/python/nvidia/dali/pipeline.py

```diff
@@ -478,6 +478,11 @@ def is_restored_from_checkpoint(self):
         """If True, this pipeline was restored from checkpoint."""
         return self._is_restored_from_checkpoint
 
+    @property
+    def num_outputs(self):
+        """Number of pipeline outputs."""
+        return self._num_outputs
```

Comment:
You need to ensure it's built first, and we just changed to having optional build, so I suggest (+ probably a test or two would be nice?)
```suggestion
        self.build()
        return self._num_outputs
```